service/matching/taskListManager.go

```diff
@@ -174,8 +174,9 @@ func newTaskListManager(
 	} else {
 		taskListTypeMetricScope = tlMgr.metricScope().Tagged(taskListDecisionTypeTag)
 	}
-	tlMgr.pollerHistory = newPollerHistory(func(state *cache.UpdatedState) {
-		taskListTypeMetricScope.UpdateGauge(metrics.PollerPerTaskListCounter, float64(state.NewSize))
+	tlMgr.pollerHistory = newPollerHistory(func() {
+		taskListTypeMetricScope.UpdateGauge(metrics.PollerPerTaskListCounter,
+			float64(len(tlMgr.pollerHistory.getAllPollerInfo())))
```

Comment:
sadly we can't (?) use history.Size() here

Reply:
yes, as discussed offline, for current cache implementation we need to evict the expired pollers everytime now.